packages/cosmic-swingset/src/block-manager.js

```diff
@@ -20,6 +21,66 @@ const END_BLOCK_SPIN_MS = process.env.END_BLOCK_SPIN_MS
   ? parseInt(process.env.END_BLOCK_SPIN_MS, 10)
   : 0;
 
+const stringToNat = s => {
+  assert.typeof(s, 'string', X`${s} must be a string`);
+  const bint = BigInt(s);
+  const nat = Nat(bint);
+  assert.equal(
+    `${nat}`,
+    s,
+    X`${s} must be the canonical representation of ${nat}`,
+  );
+  return nat;
+};
+
+// Map the SwingSet parameters to a deterministic data structure.
+//
+// NOTE: `params` is JSON-marshalled by Protobuf 3, so it is already
+// deterministic (though maybe ordered differently than alphabetical).
+const parseParams = params => {
+  const {
+    beans_per_unit: rawBeansPerUnit,
+    fee_unit_price: rawFeeUnitPrice,
+  } = params;
+  assert.equal(
+    Object(rawBeansPerUnit),
+    rawBeansPerUnit,
+    X`beansPerUnit must be an object, not ${rawBeansPerUnit}`,
+  );
+  const beansPerUnit = Object.fromEntries(
```

Comment:
I'm thinking `beansPerUnit` might read better as `beanPrices` or `swingsetPrices` or just `prices`, because the "unit" here is easy to confuse with the "unit" in "feeUnitPrice", which is something entirely different (IIUC).